noark-extraction-validator/src/main/java/com/documaster/validator/validation/Noark5Validator.java

```diff
@@ -131,6 +131,8 @@ public ValidationCollector run() throws Exception {
 				Storage.get().stopWriter();
 				Storage.get().destroy();
 			}
+
+			FileUtils.deleteQuietly(new File(getCommand().getStorageConfiguration().getDatabaseDirLocation()));
```

Comment:
Theoretically, generateReports and the Storage operations could throw an exception (an Error even; it's not like OOMs are not a good friend of this validator).

In such cases. the database file will not be deleted.

I think that this can be handled by wrapping said operations in a try and executing this in the finally block.

Alternatively, I think that the `DatabaseStorage` class can be extended so that it becomes aware of file-based databases. Then, you can perform the database file deletion in the implementation of `Storage#destroy`. This will not be perfect as an OOM coming from `generateReports` would still cause the database file to not be deleted, but the chances of this happening are much smaller with the changes that you implemented in this PR.

What do you think?

Reply:
I would go with the first solution being much safer in my opinion (in case of unforeseen Errors).

Reply:
Fixed in: https://github.com/documaster/noark-extraction-validator/pull/52/commits/58ec6cec52ab9bf3917993d34af0fc6bd05a05b3